### PR TITLE
cni: stop setting ARP proxy

### PR DESCRIPTION
**Pull request by maiqueb**
**What this PR does / why we need it**:
We've copied this part of the code from the macvlan cni
implementation, and apparently that code was a mistake - [0].

They wanted to enable arp notify, not arp proxy.

[0] - https://github.com/containernetworking/plugins/commit/e31cd2ce1a251bed484c579cf1a1d35c30c7a26b

**Special notes for your reviewer**:
We should really check if we then need to enable the ARP notification thingy before merging this.

```release-note
Stop setting ARP proxy, mimicking macvlan's cni implementation.
```

### CODE REVIEW

**Comment by maiqueb** [comment]
/hold

Let's figure out if we need to activate the ARP notifier.

**Comment by maiqueb** [comment]
/hold cancel

Ran kubevirt's e2e tests against this image; works as expected.

File: pkg/util/netlink.go [hunk 1]
```diff
@@ -316,21 +309,6 @@ func renameInterface(currentIface netlink.Link, newIfaceName string) (netlink.Li
 	return renamedMacvtapIface, nil
 }
 
-func configureArp(ifaceName string) error {
-	// For sysctl, dots are replaced with forward slashes
-	ifaceNameAllowingDots := strings.Replace(ifaceName, ".", "/", -1)
-
-	// TODO: duplicate following lines for ipv6 support, when it will be added in other places
-	ipv4SysctlValueName := fmt.Sprintf(IPv4InterfaceArpProxySysctlTemplate, ifaceNameAllowingDots)
```
Comment (phoracek) [inline pkg/util/netlink.go:324]: I think we can remove the constant now?

**Review by phoracek — CHANGES_REQUESTED** [review]
Sounds good, just a minor comment

File: pkg/util/netlink.go (hunk 1 above)
Comment (maiqueb) [reply pkg/util/netlink.go:324]: Right, it is no longer used.

**Review by phoracek — APPROVED** [review]
/lgtm

**Pull request merged**